src/coreclr/src/pal/src/thread/thread.cpp

```diff
@@ -1641,6 +1641,13 @@ CorUnix::InternalSetThreadDescription(
 
     pTargetThread->Lock(pThread);
 
+    // Ignore requests to set the main thread name because
+    // it causes the value returned by Process.ProcessName to change.
+    if (pTargetThread->GetThreadId() == getpid())
```

Comment:
```suggestion
    if (pTargetThread->GetThreadId() == (unsigned)getpid())
```
will fix the GCC error.

Reply:
I've added a `pid_t` cast on `GetThreadId`.